include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -151,18 +141,25 @@ struct __parallel_merge_submitter<_IdType, __internal::__optional_kernel_name<_N
     auto
     operator()(_ExecutionPolicy&& __exec, _Range1&& __rng1, _Range2&& __rng2, _Range3&& __rng3, _Compare __comp) const
     {
-        using _Size = oneapi::dpl::__internal::__difference_t<_Range3>;
-
-        auto __n1 = __rng1.size();
-        auto __n2 = __rng2.size();
+        const _IdType __n1 = __rng1.size();
+        const _IdType __n2 = __rng2.size();
+        const _IdType __n = __n1 + __n2;
 
         assert(__n1 > 0 || __n2 > 0);
 
         _PRINT_INFO_IN_DEBUG_MODE(__exec);
 
-        const std::uint8_t __chunk = __exec.queue().get_device().is_cpu() ? 128 : 4;
-        const _Size __n = __n1 + __n2;
-        const _Size __steps = oneapi::dpl::__internal::__dpl_ceiling_div(__n, __chunk);
+        // Empirical number of values to process per work-item
+        std::uint16_t __chunk = 128;
+        if (__exec.queue().get_device().is_gpu())
+        {
+            if (__n >= 16'777'216)
```

Comment:
1) `if (__n >= (1<<24)) //16M` probably more readable? 
2) Does it make sense to increase chunk size more in case __n is increasing ? > 32M > 64M and so on ?